### PR TITLE
Fix clippy warnings, update Cargo.toml versions

### PR DESCRIPTION
**Pull request by SomeoneToIgnore**
In the `cargo xtask lint` ouptut, there were two interesting Clippy warnings that might be interesting to investigate further:

* warning: this argument (4 byte) is passed by reference, but would be more efficient if passed by value (limit: 8 byte)

* warning: large size difference between variants

### CODE REVIEW

File: crates/ra_assists/src/handlers/add_custom_impl.rs [hunk 1]
```diff
@@ -86,7 +86,7 @@ pub(crate) fn add_custom_impl(ctx: AssistCtx) -> Option<Assist> {
                 .next_sibling_or_token()
                 .filter(|t| t.kind() == WHITESPACE)
                 .map(|t| t.text_range())
-                .unwrap_or(TextRange::from_to(TextUnit::from(0), TextUnit::from(0)));
+                .unwrap_or_else(|| TextRange::from_to(TextUnit::from(0), TextUnit::from(0)));
```
Comment (matklad) [inline crates/ra_assists/src/handlers/add_custom_impl.rs]: This is false-positive

File: crates/ra_project_model/src/lib.rs [hunk 2]
```diff
@@ -279,11 +279,14 @@ impl ProjectWorkspace {
                             }
                             if tgt.is_proc_macro(&cargo) {
                                 if let Some(proc_macro) = libproc_macro {
-                                    if let Err(_) = crate_graph.add_dep(
```
Comment (matklad) [inline crates/ra_project_model/src/lib.rs]: I'd left this in `if let Err(_)` shape, as it better shows that we ignore the error (`_`)

**Comment by SomeoneToIgnore** [comment]
Clippy warnings fix stats:

Before: 386 warnings
After: 192 warnings

**Comment by SomeoneToIgnore** [comment]
We can suppress clippy warnings for those explicitly.
For instance, there's a dangerous one that gets "fixed" by clippy and breaks RA, so there's definitely a need for some suppression.

**Comment by matklad** [comment]
string -1 on supressing them in code, +1 for supressing them in `cargo xtask lint` command

**Pull request merged**